site/tests/app/controllers/admin/SqlToolboxControllerTester.php

```diff
@@ -41,6 +41,12 @@ public function tearDown(): void {
     }
 
     public function testShowToolbox(): void {
+        $this->setUpDatabase();
+
+        /*$courseDb = $this->core->getCourseDB();
+        $courseDb->expects($this->once())->method('query')->with("SELECT * FROM information_schema.columns WHERE table_schema='public'");
+        $courseDb->expects($this->once())->method('rows')->with()->willReturn();*/
```

Comment:
> The assertEmpty on line 54 (my version) should be changed to an assertSame but should I just use similar code to what I have in my controller to test it against? Or what should I be checking it against?

For this to work, as you're using `getCourseQueries`, you'll want to mock that, and not the course db. While you could do the latter, that expands the scope of the unit under test, which we want to avoid, making the test as targeted as possible against the smallest possible unit.

```
$tables = [
	// some snippet of data as expected in prod
];
$mock_queries = $this->createMock(DatabaseQueries::class);
$mock_queries->expects($this->once())->method('getCourseSchemaTables')->willReturn($tables);
$core->setQueries($mock_queries);
```

and then below, we can confirm that our table object was passed to our WebResponse parameters:

```
$this->assertSame($response->parameters, [$tables]);
```

Reply:
Before I do the assertSame, I would need to run my array merger so the assertion can compare exactly, right? That is what I put in my latest commit which led it to pass the assertion in php unit. Or should I know the result of my array merger and hardcode that in to test against that?

Reply:
Got it. Thanks.